backend/services/vectorService.ts

```
@@ -0,0 +1,271 @@
+
+            // Get id and type from map
+            const searchResultInfo = labels.map((faissInternalIndex: number) => this.faissIdMap.get(faissInternalIndex))
+                                           .filter(info => info !== undefined) as { id: string, type: 'post' | 'reply' }[];
+            const searchResultScores: number[] = distances;
+
+            // Combine IDs, types, and scores, ensuring arrays align
+            const combinedResults = searchResultInfo.map((info, i) => ({ // Use info object
+                id: info.id,
+                type: info.type,
+                score: searchResultScores[i] !== undefined ? searchResultScores[i] : Infinity // Handle potential misalignment
+            })).filter(r => r.id && isFinite(r.score));
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)